### PR TITLE
Discover dependencies table version automatically

### LINKED ISSUE #1363: Dependencies V2 not friendly for new users
## Requirement - what kind of business use case are you trying to solve?

Make dependencies v2 more new user friendly.

## Problem - what in Jaeger blocks you from solving the requirement?

#1328 added a new schema mapping for dependencies_v2 with is by default used by cassandra-create-schema scripts, however the v2 table is not by default used by jaeger (you have to pass in a flag). This means that new users will not be able to query dependencies by default.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Do a DESC query on the cassandra jaeger keyspace on startup and see if the dependencies_v2 table is available. If so, use v2, else use v1. 

## Any open questions to address

How will this work with the flag? Should we just go ahead and deprecate it?

### PR DESCRIPTION
**Pull request by black-adder**
Signed-off-by: Won Jun Jang <wjang@uber.com>

Resolves #1363

- [ ] Deprecate `EnableDependenciesV2` flag

### CODE REVIEW

File: plugin/storage/cassandra/dependencystore/bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+package dependencystore
+
+import (
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+)
+
+// GetDependencyVersion attempts to determine the version of the dependencies table.
+// TODO: Remove this once we've migrated to V2 permanently.
+func GetDependencyVersion(s cassandra.Session) Version {
+	if err := s.Query("SELECT ts from dependencies_v2 limit 1;").Exec(); err != nil {
```
Comment (yurishkuro) [inline plugin/storage/cassandra/dependencystore/bootstrap.go]: what kind of error is returned? just curious

Comment (black-adder) [reply plugin/storage/cassandra/dependencystore/bootstrap.go]: `InvalidRequest: Error from server: code=2200 [Invalid query] message="unconfigured table dependencies"`

**Comment by pavolloffay** [comment]
@black-adder I have raised this also because of https://github.com/jaegertracing/spark-dependencies/issues/58. There can be other components which expect schema for v1 dependencies. We have failing CI jobs now in opearator bc dep. job tries to write to old dependnecies table.

in spark job we should do the same - automatically detect the table

File: plugin/storage/cassandra/dependencystore/bootstrap.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// limitations under the License.
+
+package dependencystore
+
+import (
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+)
+
+// GetDependencyVersion attempts to determine the version of the dependencies table.
+// TODO: Remove this once we've migrated to V2 permanently.
```
Comment (pavolloffay) [inline plugin/storage/cassandra/dependencystore/bootstrap.go]: Add a pointer to issue

**Comment by pavolloffay** [comment]
@black-adder can this be merged?

**Pull request merged**